src/nexus.h

```diff
@@ -41,18 +41,27 @@ class Nexus : public QObject
 {
     Q_OBJECT
 public:
-    void start();
-    void showMainGUI();
-
     static Nexus& getInstance();
     static void destroyInstance();
-    static Core* getCore();
-    static Profile* getProfile();
-    static void setProfile(Profile* profile);
-    static Widget* getDesktopGUI();
     static QString getSupportedImageFilter();
     static bool tryRemoveFile(const QString& filepath);
 
+    void start();
+    void showMainGUI();
+
+public:
+    inline Widget* getDesktopGUI() const
+    {
+        return widget;
+    }
+
+    inline Profile* getProfile() const
```

Comment:
@Diadlo The correct question is: Why don't we inline getters in general? :smile:

Basically there's two approaches:
1. No PIMPL => getters **should be** inlined in general
2. PIMPL => getters cannot be inline'd (due to the nature of a private class residing in implementation)

Inline means nothing but compiling the function body into the code, which is (tada!!) a pointer or reference. However, if the compile algorithm finds it is not possible, it won't do it anyway and add use a function pointer instead. Btw. you won't find much inline'd getters in the Qt internal framework, as the concept is highly pimp(l)'ed. :smile: